### ISSUE TITLE
Rolling statistics not working

### ISSUE THREAD

**Issue by rgaudin** [SEL-Columbia/bamboo#487]
There are two problems with rolling statistics:
- The doc is unclear about how to use it but I'm not sure how to fix it. 
- It's not working.
  I get a 500 error while using the example provided (which should work according to the code).

On my local setup, I have the following trace 

```
File "/Volumes/data/src/envs/bamboo/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/Volumes/data/src/envs/bamboo/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/Volumes/data/src/envs/bamboo/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "bamboo/controllers/datasets.py", line 413, in rolling
    content_type=content_type)
  File "bamboo/controllers/abstract_controller.py", line 119, in _safe_get_and_call
    result = action(dataset)
  File "bamboo/controllers/datasets.py", line 409, in action
    dframe = dataset.rolling(win_type, int(window))
  File "bamboo/models/dataset.py", line 618, in rolling
    return BambooFrame(rolling_window(dframe, window, win_type))
  File "/Volumes/data/src/envs/bamboo/lib/python2.7/site-packages/pandas/stats/moments.py", line 644, in rolling_window
    window = sig.get_window(win_type, window).astype(float)
  File "/Volumes/data/src/envs/bamboo/lib/python2.7/site-packages/scipy/signal/windows.py", line 462, in get_window
    if winstr in ['blackman', 'black', 'blk']:
UnboundLocalError: local variable 'winstr' referenced before assignment
```

Please not that I am running `scipy==0.11.0` as the `0.9.0` refuses to install on OSX (confirmed bug).
I don't know if it's related but since it fails also on bamboo.io, I believe there's a problem anyway.

**Comment by pld** [comment]
Intriguing, this test might help explain, https://github.com/modilabs/bamboo/blob/master/bamboo/tests/controllers/test_datasets_ts_functions.py#L86

**Comment by rgaudin** [comment]
Thanks for the pointer ; When not passing the `win_type` parameter, it works (like in your test).

I checked the code and it seems that an `str` type is expected while I believe cherryPy provides `unicode`.

Fixed by ce4d9df7337cdb0cd19056085a061256d5ccd930

**Comment by pld** [comment]
awesome!
